climpred/bootstrap.py

```diff
@@ -393,6 +394,99 @@ def varweighted_mean_period_threshold(control, sig=95, iterations=500, time_dim=
     )
 
 
+def _bootstrap_hindcast_over_init_dim(
+    hind,
+    hist,
+    verif,
+    dim,
+    resample_dim,
+    iterations,
+    metric,
+    comparison,
+    alignment,
+    compute,
+    reference_compute,
+    resample_uninit,
+    reference_alignment,
+    **metric_kwargs,
+):
+    """Helper function for looping bootstrap skill.
+    Needed for hindcast when resampling over init dimension."""
+    pers_skill = []
+    bootstrapped_init_skill = []
+    bootstrapped_uninit_skill = []
+    # reset inits when probabilistic, otherwise tests fail
+    for i in range(iterations):
+        # resample with replacement
+        smp_hind = _resample(hind, resample_dim)
+        # compute init skill
+        init_skill = compute(
+            smp_hind,
+            verif,
+            metric=metric,
+            comparison=comparison,
+            alignment=alignment,
+            add_attrs=False,
+            dim=dim,
+            **metric_kwargs,
+        )
+        # reset inits when probabilistic, otherwise tests fail
+        if (
+            resample_dim == 'init'
+            and metric.probabilistic
+            and 'init' in init_skill.coords
+        ):
+            init_skill['init'] = hind.init.values
+        bootstrapped_init_skill.append(init_skill)
+        # generate uninitialized ensemble from hist
+        uninit_hind = resample_uninit(hind, hist)
```

Comment:
Just a thought. This method takes an uininitialized ensemble (member, time) and randomly selects members at each `init` of `hind` to create what looks like an initialized forecast of (member, lead, init). In short, this is resampling over the `member` dimension. 

Would it be more appropriate here to somehow resample the "inits" of the uninitialized ensemble for comparison? Maybe first make this object, and then resample the inits in the same way as the initialized ensemble?

This is scope creep for the current PR. But something to think about for the future.

Reply:
I think only about uninit members becasethereis no common initialisation as in init. Therefore I justresample uninit over member